src/DataFrame/IO/Parquet.hs

```diff
@@ -240,7 +240,9 @@ processColumnPages (maxDef, maxRep) pages pType _ maybeTypeLength = do
                     ERLE_DICTIONARY -> decodeDictV1 dictValsM maxDef defLvls nPresent afterLvls
                     EPLAIN_DICTIONARY -> decodeDictV1 dictValsM maxDef defLvls nPresent afterLvls
                     other -> error ("Unsupported v2 encoding: " ++ show other)
-
+            DictionaryPageHeader{} -> undefined
+            INDEX_PAGE_HEADER -> undefined
```

Comment:
Index page headers are unimplemented in the parquet spec. So fine to leave this as undefined.

https://github.com/apache/parquet-format/blob/c3f7be7dc615de76910154d690c534768ba25d6f/src/main/thrift/parquet.thrift#L694